File: Python/linked-list/design_hashmap.py
class MyHashMap:

    def __init__(self):
        """
        Initialize your data structure here.
        """
        self.remain = [[] for _ in range(769)]

    def put(self, key: int, value: int) -> None:
        """
        value will always be non-negative.
        """
        k = key % 769
        for i in range(len(self.remain[k])):
            if self.remain[k][i][0] == key:
                self.remain[k][i][1] = value
                return
        self.remain[k].append([key, value])


    def get(self, key: int) -> int:
        """
        Returns the value to which the specified key is mapped, or -1 if this map contains no mapping for the key
        """
        k = key % 769
        for i in range(len(self.remain[k])):
            if self.remain[k][i][0] == key:
                return self.remain[k][i][1]
        
        return -1


    def remove(self, key: int) -> None:
        """
        Removes the mapping of the specified value key if this map contains a mapping for the key
        """
        k = key % 769
        for i in range(len(self.remain[k])):
            if self.remain[k][i][0] == key:
                del self.remain[k][i]
                break

File: core/tmp/Java/src/com/dbc/algorithm/LinkedList/DesignHashMap.java
package com.dbc.algorithm.LinkedList;

import java.util.Iterator;
import java.util.LinkedList;

public class DesignHashMap {
    private LinkedList[] remain;

    public DesignHashMap() {
        remain = new LinkedList[769];
        for (int i = 0; i < 769; i++) {
            remain[i] = new LinkedList<Integer[]>();
        }
    }

    /**
     * value will always be non-negative.
     */
    public void put(int key, int value) {
        int k = key % 769;
        Iterator<Integer[]> iterator = remain[k].iterator();
        while (iterator.hasNext()) {
            Integer[] element = iterator.next();
            if (element[0] == key) {
                element[1] = value;
                return;
            }
        }
        remain[k].offerLast(new Integer[]{key, value});
    }

    /**
     * Returns the value to which the specified key is mapped, or -1 if this map contains no mapping for the key
     */
    public int get(int key) {
        int k = key % 769;
        Iterator<Integer[]> iterator = remain[k].iterator();
        while (iterator.hasNext()) {
            Integer[] element = iterator.next();
            if (element[0] == key) {
                return element[1];
            }
        }
        return -1;
    }

    /**
     * Removes the mapping of the specified value key if this map contains a mapping for the key
     */
    public void remove(int key) {
        int k = key % 769;
        Iterator<Integer[]> iterator = remain[k].iterator();
        while (iterator.hasNext()) {
            Integer[] element = iterator.next();
            if (element[0] == key) {
                remain[k].remove(element);
                return;
            }
        }
    }
}
